### PR TITLE
Add a few optimisations to speed up the display of the order history page and to allow potential override to skip unneeded costly informations

### PR DESCRIPTION
**Pull request by jocel1**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Speed up order page. Allow more efficient override
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Check the order history page still work properly

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8469)

### CODE REVIEW

**Comment by jocel1** [comment]
I don't have the blackfire profile anymore, but it helps a customer reducing the order history load page from well.. a timeout to a few second for a specific profile loaded with orders. So I would say the gain is 90%+ for them

**Comment by LittleBigDev** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[controllers/front/HistoryController.php, line 47 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8469#-Kyzyzgh4mogB32dvU_G:-Kyzyzgi6GPVbbUZ6T24:belup5u) ([raw file](https://github.com/prestashop/prestashop/blob/94bd16b082475a3684fe7254971461c6863c001c/controllers/front/HistoryController.php#L47)):*
> ```PHP
>         }
> 
>         if ($this->order_presenter === null) {
> ```

Yoda condition ?
Personal taste, I would rather use `isset()` (basically the same, but easier to read)

---

*[src/Adapter/Order/OrderPresenter.php, line 295 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8469#-KyzynFSNtnh2UVF_wXk:-KyzynFSNtnh2UVF_wXl:ba4lkro) ([raw file](https://github.com/prestashop/prestashop/blob/94bd16b082475a3684fe7254971461c6863c001c/src/Adapter/Order/OrderPresenter.php#L295)):*
> ```PHP
>     /**
>      * @param Order $order
>      * @param bool  $getShippingInfos
> ```

Not matching the real parameter name

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8469)*

**Review by LittleBigDev — CHANGES_REQUESTED** [review]
Thank you @jocel1 !
See my (few) comments above.

**Comment by jocel1** [comment]
fixed!

**Comment by LittleBigDev** [comment]
Nice
:lgtm:

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8469#-:-L-63fLG9vY4bxpjsrd8:b5w74d0)*

**Comment by LittleBigDev** [comment]
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8469)*

**Review by LittleBigDev — DISMISSED** [review]

**Comment by marionf** [comment]
Hi @jocel1 

I have a notice on the order confirmation step 
capture du 2018-01-10 11-50-35

The same notice on the order history
capture du 2018-01-10 11-52-35

Another notice on the order detail page
capture du 2018-01-10 11-57-03

**Comment by jocel1** [comment]
@marionf this should be fixed now, thanks!

**Comment by LittleBigDev** [comment]
Reviewed 4 of 5 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/prestashop/prestashop/8469)*

**Review by LittleBigDev — DISMISSED** [review]

File: classes/Cart.php [hunk 1]
```diff
@@ -3865,23 +3865,34 @@ public function isVirtualCart()
         }
 
         if (!isset(self::$_isVirtualCart[$this->id])) {
-            $products = $this->getProducts();
-            if (!count($products)) {
-                return false;
+            $isVirtual = true;
```
Comment (mickaelandrieu) [inline classes/Cart.php]: How about ...

```php
$isVirtual = !$this->hasRealProducts();
```

Comment (jocel1) [reply classes/Cart.php]: brilliant! 👍

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thanks @jocel1 !

**Pull request merged**